### PR TITLE
Merged and reworked episode 03

### PR DESCRIPTION
**Pull request by anenadic**
Merged Anne's and mine changes in episode 03, but then pulled out my bits on website configuration and simply pasted in episode 05 on a [separate branch](https://github.com/carpentries-incubator/building-websites-with-jekyll-and-github-or-gitlab/tree/episode-05) (which will now need reworking too). I have tested the code yet - this will need reviewing once again, when we have most of the content and the whole narrative ready.

### CODE REVIEW

File: _episodes/03-github-pages.md [hunk 1]
```diff
@@ -3,275 +3,126 @@ title: "Hosting Pages and Websites on GitHub"
...
+Let's continue from the GitHub repository we have created in the previous episode. One important file you should have is `index.md`: it will become the homepage of your project website.
 
-## Turn on GitHub Pages for your main branch
+## Turning on GitHub Pages for Your Repository
 
-GitHub Pages uses a special branch in your GitHub repository to look for website content, and by default this branch is named `gh-pages`. 
+There are two approaches to tell GitHub your repository is hosting a website and configure GitHub pages to render it. 
 
-It is a convention used by Github and we need to change this default behavior because the content of our repository is under a branch called *main*.
+The first one is to create a new branch called `gh-pages`. By convention, GitHub Pages uses this branch to look for the website content and by creating it you implicitly tell GitHub that you want your content published. Once you can create `gh-pages` from your current branch (`main`, created by default when you created the repository), you can then choose to delete the branch `main` to avoid any confusion.
```
Comment (annefou) [inline _episodes/03-github-pages.md]: I wonder if it makes sense to talk about gh-pages at this stage. 

Wouldn't be simpler to show how to select `main` branch to "create" a website and than have a called out about gh-pages.

**Review by annefou — COMMENTED** [review]
All other changes are great. So I don't know what you feel about my comment about `gh-pgaes` but maybe we can merge fierst and re-discuss it later.

File: _episodes/03-github-pages.md (hunk 1 above)
Comment (anenadic) [reply _episodes/03-github-pages.md]: Yes, might have got things backward - I tell them about gh-pages and then go and explain the approach which is actually less favourable. I'll fix this tomorrow as per your suggestion @annefou.

Comment (anenadic) [reply _episodes/03-github-pages.md]: OK, I fixed the text to address this comment @annefou via ec981b4891081876fffcc766ddc2ba1439c0226b.

File: _episodes/03-github-pages.md [hunk 2]
```diff
@@ -3,275 +3,126 @@ title: "Hosting Pages and Websites on GitHub"
...
->
-{: .callout}
-
-Let's start from the github repository we have created in the previous episode {{ site.project-pages }}. One important file you should have is `index.md`: it will become the homepage of your project website.
-
-In this episode, you will learn how to turn your project repository into a website.
+There are other services available to create and publish website but one of the main advantages of GitHub Pages is that you can version control your website and therefore keep track of all your changes. This is particularly helpful for collaborating on a project website. [GitLab](https://about.gitlab.com/) offers very similar services but GitHub pages is the simplest approach.
 
-## Turn on GitHub Pages for your main branch
+Let's continue from the GitHub repository we have created in the previous episode. One important file you should have already is `index.md`: it will become the homepage of your project website.
```
Comment (tobyhodges) [inline _episodes/03-github-pages.md]: I think the learners will only have the `README.md` file at the starting point of this lesson. In the absence of a file called `index.md`, the README is used as the landing page of the GitHub Pages site, so it shouldn't change anything for this section, except adding a part to create a separate `index.md` later...

File: _episodes/03-github-pages.md [hunk 3]
```diff
@@ -3,275 +3,126 @@ title: "Hosting Pages and Websites on GitHub"
...
+    ![Project website URL](../fig/website_url.png)
+    
+4. It may take a while for GitHub (from a few seconds to a few minutes) to compile your website (depending on GitHub's availability and the complexity of 
+your website) and it may not become visible immediately. You will know it is ready when the link appears in green box with a "tick" in front of the web address (as shown in the figure below). 
 
-- Once GitHub pages are built, open the link (hilighted green box): your default browser will open and show your project website.
+    ![Project website URL - successfull build](../fig/website_url_green_tick.png)
 
-![Default project website](../fig/default_website.png)
+5. Once ready, you should see the contents of the `index.md` file that we created earlier, rendered as a website.
```
Comment (tobyhodges) [inline _episodes/03-github-pages.md]:
```suggestion
5. Once ready, you should see the contents of the `README.md` file that we created earlier, rendered as a website. 
```

File: _episodes/03-github-pages.md [hunk 4]
```diff
@@ -3,275 +3,126 @@ title: "Hosting Pages and Websites on GitHub"
...
+You can edit the details of your repository by clicking on the little cog button as shown on the figure below.
 
 ![Repository details](../fig/repository_details.png)
 
-> ## Add new content to your website
->
-> Add a new section **About** in `index.md`. You can edit `index.md` from the GitHub interface and directly commit your changes.
-> Then view your changes on the website.
+By doing this, you add a link to the repository's website on your repository's landing page and anyone (including yourself) can access it quickly when visiting your GitHub repository.
+
```
Comment (tobyhodges) [inline _episodes/03-github-pages.md:102]:
```suggestion

Up to now, the content of your webpage is identical to what visitors to your repository on GitHub will see in the project's `README.md` file. It is often better to have different content in the homepage of your site - aimed at visitors to the website - than in the README, which is supposed to provide information about the GitHub repository e.g. license information, local installation instructions, the structure and configuration of the repository, etc. By default, the homepage for a GitHub Pages website is built from a file called `index.md`: in the absence of a file with that name the "fallback" option is to use `README.md`, which is why your homepage is currently being built from that file.

To separate the contents of the repository's README from the website's homepage, create a new file called `index.md` and copy the contents of the README into that file. We'll add more content to that file in the next exercise.
```

Proposing this to bridge the gap between README and `index.md` for the homepage. Feel free to overwrite/ignore any of it!

**Review by tobyhodges — CHANGES_REQUESTED** [review]
At the sprint yesterday, I think we concluded that all exercises in the Markdown episode would be done in `README.md`, meaning there won't yet be an `index.md` file by the beginning of this episode. (@JulianKarlBauer can confirm...) I've suggested an additional paragraph to bridge this gap, but you might want to do it differently...

File: _episodes/03-github-pages.md (hunk 2 above)
Comment (anenadic) [reply _episodes/03-github-pages.md]: Oh, I've missed that bit of discussion and have just added exercises that work with `index.md`. It's not a biggie, I will  add a bit where we create it.

File: _episodes/03-github-pages.md (hunk 3 above)
Comment (anenadic) [reply _episodes/03-github-pages.md]: Fixed via 596a28b.

File: _episodes/03-github-pages.md (hunk 2 above)
Comment (anenadic) [reply _episodes/03-github-pages.md]: Fixed via 596a28b.

**Comment by anenadic** [comment]
The episode is ready for review once again - I have added solutions to exercises in the meantime and addressed the fact that we only have README.md at the start of the episode @tobyhodges @annefou.

**Review by tobyhodges — APPROVED** [review]

**Comment by tobyhodges** [comment]
Looks great, thank you @anenadic

File: _episodes/03-github-pages.md [hunk 5]
```diff
@@ -3,275 +3,182 @@ title: "Hosting Pages and Websites on GitHub"
...
+- "You can configure any branch of a repository to be used for website (it does not have to be `gh-pages`)"
+- "GitHub publishes websites on special URLs formatted as 'https://GITHUB_USERNAME.github.io/REPOSITORY_NAME'"
+
 ---
 
-# What is GitHub Pages?
+Now that you know how to create Markdown files, let's see how to turn them into web pages. GitHub has a 
+service just for that called GitHub Pages.  
+
+# Publishing a Repository as a Website With GitHub Pages
```
Comment (tobyhodges) [inline _episodes/03-github-pages.md]:
```suggestion
# Publishing a Website with GitHub Pages
```

File: _episodes/03-github-pages.md [hunk 6]
```diff
@@ -3,275 +3,182 @@ title: "Hosting Pages and Websites on GitHub"
...
+There are other services available to create and publish website but one of the main advantages of GitHub Pages is that you can version control your website and therefore keep track of all your changes. This is particularly helpful for collaborating on a project website. [GitLab](https://about.gitlab.com/) offers very similar services but GitHub pages is the simplest approach.
 
-GitHub Pages is a service from GitHub to publish a website directly from a Git repository.
+Let's continue from the GitHub repository we have created in the previous episode. One important file you should already have is `README.md`, which will become the homepage of your project website (until we add the index file later on).
 
-> ## Why Github?
->
-> There are other services available to create and publish website but one of the main advantage of GitHub Pages is that you can version control your website and therefore keep track of all your changes. This is particularly helpful for collaborating on a project website. [GitLab](https://about.gitlab.com/) offers very similar services but GitHub pages is the simplest approach.
+## Turning on GitHub Pages for Your Repository
```
Comment (tobyhodges) [inline _episodes/03-github-pages.md]:
```suggestion
## Enabling GitHub Pages
```

File: _episodes/03-github-pages.md [hunk 7]
```diff
@@ -3,275 +3,182 @@ title: "Hosting Pages and Websites on GitHub"
...
+convention gives us a way
+of neatly creating Web address for just about any GitHub repository without any conflicts.
 
-- Once GitHub pages are built, open the link (hilighted green box): your default browser will open and show your project website.
+> ## Customising domain
+> **TODO**
+{: .callout}
 
-![Default project website](../fig/default_website.png)
+## Updating Repository Details With Website URL
```
Comment (tobyhodges) [inline _episodes/03-github-pages.md]:
```suggestion
## Making Your Pages More Findable
```

**Review by tobyhodges — COMMENTED** [review]
Putting some suggestions here for the headings, to address #57. Happy to prepare these in a separate PR if you prefer?

**Comment by anenadic** [comment]
> Putting some suggestions here for the headings, to address #57. Happy to prepare these in a separate PR if you prefer?

All fab, thank you! Accepted them and will close the issue #57. Ready for merge I'd say @tobyhodges ?

**Comment by tobyhodges** [comment]
Thanks @anenadic 😀

**Pull request merged**